### DISCUSSION TITLE
Cant define enum type in IDL project - .idl file

### DISCUSSION THREAD

**Discussion by edwin98thompson** [jmmorato/openddsharp#207] [Q&A] [answered]
Attempting to define an IDL which features an enum as a field and project wont compile. Project configuration is not an issue as build succeeds when only the long data type is present. I believe I am defining the enum as per the documentation.

Code snippet as shown:

![image](https://github.com/jmmorato/openddsharp/assets/68071932/d3b04051-fbc3-4413-a4a4-53949d4a582d)

Example of some of the errors:

![image](https://github.com/jmmorato/openddsharp/assets/68071932/d7618fcd-90c7-4436-a30b-2592f1831f15)

Any help would be much appreciated, thanks.

**Comment by jmmorato** [comment, marked as answer]
Your issue is not the definition of the enum in the IDL but the field `IGNORE` defined inside... `IGNORE` is a constant defined in `WinBase.h` and the compiler complains about in the enum definition.

It is happening only in Windows and in the code generated by OpenDDS so I don't think we can do much about.

Can you change the name to `STATE_IGNORE` for example and try again?

**Reply by edwin98thompson** [reply]
That fixed the problem, many thanks!
